packages/appsync-modelgen-plugin/src/utils/process-belongs-to.ts

```diff
@@ -28,7 +29,7 @@ export function processBelongsToConnection(
       `A 'belongsTo' field should match to a corresponding 'hasMany' or 'hasOne' field`
     );
   }
-  const otherSideField = isCustomPKEnabled ? otherSideConnectedFields[0] : getConnectedFieldV2(field, model, otherSide, connectionDirective.name);
+  const otherSideField = isCustomPKEnabled ? otherSideConnectedFields[0] : getConnectedFieldV2(field, model, otherSide, connectionDirective.name, false, respectReferences);
```

Comment:
The additional `false` here is for the optional argument `shouldUseModelNameFieldInHasManyAndBelongsTo` which defaults to `false`.